### DISCUSSION TITLE
Riprogrammazione firmware Zigbee Dongle plus - problema entrare in bootloader

### DISCUSSION THREAD

**Discussion by braudoo** [indomus/forum#1955] [Hardware & Firmware]
Salve a tutti, come da titolo sto cercando di riprogrammare il fw del mio Zigbbe Usb dongle plus, ma mi trovo a non poter procedere a causa del fatto che non va in bootloader.
Ho fatto entrambi le prove senza tenere tasto boot e tenendo il tasto boot.
Questi i risultati:

SENZA PREMERE TASTO BOOT:
```
pi@raspberrypi:~ $ ./ports.sh
/dev/bus/usb/001/001 - Linux_5.15.61-v8+_xhci-hcd_xHCI_Host_Controller_0000:01:00.0
/dev/bus/usb/001/011 - Samsung_M3_Portable_Samsung_M3_Portable_1CCA7DE81E0000A3
/dev/sg0 -
/dev/sda - Samsung_M3_Portable_1CCA7DE81E0000A3-0:0
/dev/sda1 - Samsung_M3_Portable_1CCA7DE81E0000A3-0:0
/dev/bsg/0:0:0:0 -
/dev/bus/usb/001/002 - 2109_USB2.0_Hub
/dev/bus/usb/001/010 - ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220719094400
/dev/ttyACM0 - ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220719094400
/dev/bus/usb/002/001 - Linux_5.15.61-v8+_xhci-hcd_xHCI_Host_Controller_0000:01:00.0


pi@raspberrypi:~/cc2538-bsl $ ./cc2538-bsl.py -p /dev/ttyACM0 -evw --bootloader-sonoff-usb CC1352P2_CC2652P_launchpad_coordinator_20220219.hex
sonoff
Opening port /dev/ttyACM0, baud 500000
Reading data from CC1352P2_CC2652P_launchpad_coordinator_20220219.hex
Your firmware looks like an Intel Hex file
Connecting to target...
ERROR: Timeout waiting for ACK/NACK after 'Synch (0x55 0x55)'
```

PREMENDO IL TASTO BOOT:
```
pi@raspberrypi:~ $ ./ports.sh
/dev/bus/usb/001/001 - Linux_5.15.61-v8+_xhci-hcd_xHCI_Host_Controller_0000:01:00.0
/dev/bus/usb/001/004 - Samsung_M3_Portable_Samsung_M3_Portable_1CCA7DE81E0000A3
/dev/sg0 -
/dev/sda - Samsung_M3_Portable_1CCA7DE81E0000A3-0:0
/dev/sda1 - Samsung_M3_Portable_1CCA7DE81E0000A3-0:0
/dev/bsg/0:0:0:0 -
/dev/bus/usb/001/002 - 2109_USB2.0_Hub
/dev/bus/usb/001/005 - 1a86_USB_Single_Serial_54DD004189
/dev/ttyACM0 - 1a86_USB_Single_Serial_54DD004189
/dev/bus/usb/002/001 - Linux_5.15.61-v8+_xhci-hcd_xHCI_Host_Controller_0000:01:00.0

pi@raspberrypi:~/cc2538-bsl $ ./cc2538-bsl.py -p /dev/ttyACM0 -evw --bootloader-sonoff-usb CC1352P2_CC2652P_launchpad_coordinator_20220219.hex
sonoff
Opening port /dev/ttyACM0, baud 500000
Reading data from CC1352P2_CC2652P_launchpad_coordinator_20220219.hex
Your firmware looks like an Intel Hex file
Connecting to target...
ERROR: device reports readiness to read but returned no data (device disconnected or multiple access on port?)

```

Qualcuno riuscirebbe ad aiutarmi a capire in cosa sbaglio?

Quasi opterei per una chiavetta difettosa.

Grazie.

**Comment by snakuzzo** [comment]
Ma per caso la chiavetta è montata nel container docker di Home Assistant o Zigbee2mqtt ?
Se è questo il caso, non puoi flashare la chiavetta mentre è in uso. Dovresti prima fermare il container che la utilizza e poi provare.
Ad ogni modo guarda [QUI](https://github.com/JelmerT/cc2538-bsl/issues?q=is%3Aissue+%220x55%22) tra le issue del progetto cercando quello che ti appare nel messaggio di errore (se stai flashando tramite RPI c'è qualcosa pure relativo a quello ed all'editing del file cmdline.txt)...

**Reply by braudoo** [reply]
Quindi dovrei fermare il container di HA prima di procedere al flash?
Premetto che avevo iniziato a fare prove prima di istanziare Z2M, poi l'ho istanziato per provare l'antenna C2531 che avevo prima.
Ad ogni modo guardo anche le issue e tentò nel capirci.
Grazie

**Reply by snakuzzo** [reply]
Il discorso è...
se la porta è in uso da HA per ZHA...devi fermare il container di HA.
se la porta è in uso da Z2M devi fermare il container di Z2M

**Reply by braudoo** [reply]
Non ho ZHA come integrazione installata ma solo Z2M.
A questo punto verifico le issue e poi eventualmente fermo il container di HA e riprovo a flasheare.

**Reply by snakuzzo** [reply]
ma scusa...ti ho detto "ferma Z2M" e prova...
se poi non va guardi tra le issue (dove comunque anche lì ti dicono di fare sta cosa).
Ad ogni modo, basta ragionarci un attimo... è proprio la logica che ti dice questo. Come pretendi di aggiornare il firmware di un device mentre è utilizzato da un'applicazione?

**Reply by braudoo** [reply]
Ancor prima di installare Z2M mi dava il problema, per questo nn mi sembra sia quello che vada in conflitto. 
Inoltro ho provato ad aggiornarlo tramite PC e lì non ci sono applicazioni che lo utilizzano.
Pero provo ad andare per step fermando Z2M e poi HA.

**Reply by snakuzzo** [reply]
HA non serve fermarlo (non andare a caso).
Quello che serve fermare è il container che tiene impegnata quella porta...e se usi Z2M, HA non c'entra assolutamente nulla.
Ad ogni modo, se riscontravi lo stesso problema, prova a verificare tra le issue...ti ho dato anche un indizio sul discorso di cmdline.txt che dovrebbe essere presente in una delle issue.
Ciao.

**Reply by braudoo** [reply]
ottimo... comincio a scavare :-)
